### ISSUE TITLE
Categorical_crossentropy and categorical_focal_crossentropy ignore axis for smoothing and warnings

### ISSUE THREAD

**Issue by buildwithsuhana** [keras-team/keras#23451]
When using categorical_crossentropy or categorical_focal_crossentropy with a custom axis (e.g., in 3D tensors with (batch, classes, spatial) format), the label_smoothing logic and the shape-related SyntaxWarning both ignore the user-provided axis and default to the last dimension (-1).

### Steps to Reproduce
```
import numpy as np
from keras.src.losses import losses

# 3D tensor: (batch, classes, spatial) -> axis=1 has 3 classes, axis=2 has 2 spatial dims
y_true = np.array([[[1, 0], [0, 1], [0, 0]]], dtype="float32") # Shape (1, 3, 2)
y_pred = np.array([[[0.8, 0.1], [0.1, 0.8], [0.1, 0.1]]], dtype="float32")

# We specify axis=1, but smoothing will incorrectly use axis=-1 (length 2)
loss = losses.categorical_crossentropy(y_true, y_pred, label_smoothing=0.1, axis=1)
```

Colab notebook with bug reproduction: https://colab.research.google.com/drive/1LyDEp9HJI-MoCOAwTfmqbEBV8ec-O9cj?resourcekey=0-IuR4p5aMkSy2FYyeEgYC_g&usp=sharing

  ### Expected Behavior
   1. Label Smoothing: Should calculate num_classes based on the specified axis.
   2. Warnings: The SyntaxWarning (for single-class inputs) should check the class counts on the specified axis, not just the last dimension.
   3. Robustness: The code should canonicalize the axis to handle negative indices and verify static shapes during tracing.

  ### Actual Behavior
  The implementation hardcodes shape[-1] for both smoothing math and warnings, leading to incorrect loss values when axis is not -1.
